### ISSUE TITLE
Not all secrets showing with SecretServer

### ISSUE THREAD

**Issue by moseala55** [royalapplications/toolbox#33]
Connected and authenticated to on prem  SecretServer just fine.  Not all secrets that I know and have verified I have access too, including secrets that I have created are showing.  "Personal Folder" does not show in the list as well.  When refreshing the dynamic folder, I do get an information error in the bottom left saying "Some tokens couldn't be resolved".

**Comment by lemonmojo** [comment]
Please post a screenshot of the full error message.
Also, there's a limit on how many entries should be returned by the API. How many are you expecting approximately?

**Comment by moseala55** [comment]
![2022-02-03_8-20-24](https://user-images.githubusercontent.com/98920092/152361041-5d90529d-9ea3-4b52-abe9-1c3ef2e13b09.png)
Screenshot of the log uploaded.  We have 5,817 items in our SecretServer.  If that is to many entries to be returned, how would I limit the script to pull from specific folders?

**Comment by st9rm1337** [comment]
Please try adjusting the value on line 103 of the script as follows:
"paging.take" = 1000;

you can also pick a different number, which suits your needs.
​

**Comment by moseala55** [comment]
Increasing the value seems to work, but at the cost of causing the application to "lock up" while it reloads the dynamic folder, especially on application start.  Is there anyway to limit the scrip to pull from only a specific folder?

**Comment by lemonmojo** [comment]
@moseala55 Yes, you can filter the secrets by certain criteria, including the folder ID.
To do so, you'll want to modify the request body that is sent to the secret server from line 102 to 104 of the dynamic folder script.

By default, the body looks like this:
```
$secretsRequestBody = @{
  "paging.take" = 1000;
}
```

To include only secrets from a specific folder ID, you would adjust those lines like so:

```
$secretsRequestBody = @{
  "paging.take" = 1000;
  "filter.folderId" = 123;
}
```

For more information on the supported filter criteria, please see Thycotic's [Rest API documentation](https://updates.thycotic.net/secretserver/restapiguide/10.8/TokenAuth/#operation--secrets-get).

**Comment by jpitpol** [comment]
Also, the "filter.folderId" = 123; does not work.
